Detectors/TPC/workflow/src/FileReaderWorkflow.cxx

```diff
@@ -0,0 +1,132 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   FileReaderWorkflow.cxx
+
+#include "TPCWorkflow/RecoWorkflow.h"
+#include "TPCWorkflow/PublisherSpec.h"
+#include "TPCWorkflow/TPCSectorCompletionPolicy.h"
+
+#include "SimulationDataFormat/IOMCTruthContainerView.h"
+#include "SimulationDataFormat/ConstMCTruthContainer.h"
+#include "SimulationDataFormat/MCCompLabel.h"
+
+#include "TPCWorkflow/TrackReaderSpec.h"
+
+// we need a global variable to propagate the type the message dispatching of the
+// publisher will trigger on. This is dependent on the input type
+o2::framework::Output gDispatchTrigger{"", ""};
+
+// Global variable used to transport data to the completion policy
+o2::tpc::reco_workflow::CompletionPolicyData gPolicyData;
+
+// add workflow options, note that customization needs to be declared before
+// including Framework/runDataProcessing
+void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
+{
+  using namespace o2::framework;
+
+  std::vector<ConfigParamSpec> options{
+    {"input-type", VariantType::String, "clusters", {"clusters"}},
+    {"dispatching-mode", VariantType::String, "prompt", {"determines when to dispatch: prompt, complete"}},
+    {"disable-mc", VariantType::Bool, false, {"disable sending of MC information"}}};
+
+  std::swap(workflowOptions, options);
+}
+
+// customize dispatch policy, dispatch immediately what is ready
+void customize(std::vector<o2::framework::DispatchPolicy>& policies)
+{
+  using DispatchOp = o2::framework::DispatchPolicy::DispatchOp;
+  // we customize all devices to dispatch data immediately
+  auto readerMatcher = [](auto const& spec) {
+    return std::regex_match(spec.name.begin(), spec.name.end(), std::regex(".*-reader"));
+  };
+  auto triggerMatcher = [](auto const& query) {
+    // a bit of a hack but we want this to be configurable from the command line,
+    // however DispatchPolicy is inserted before all other setup. Triggering depending
+    // on the global variable set from the command line option. If scheduled messages
+    // are not triggered they are sent out at the end of the computation
+    return gDispatchTrigger.origin == query.origin && gDispatchTrigger.description == query.description;
+  };
+  policies.push_back({"prompt-for-reader", readerMatcher, DispatchOp::WhenReady, triggerMatcher});
+}
```

Comment:
Is this customisation and the ``dispatching-mode`` option really needed? It should be enough to get the ``input-type`` and ``disable-mc`` options and invoke one of both of these lines:
https://github.com/AliceO2Group/AliceO2/blob/06240bc05edf6eb41c624fb3cfcaaf7ebe0aaa8e/Detectors/GlobalTrackingWorkflow/src/MatchTPCITSWorkflow.cxx#L45-L55

Reply:
Not really needed, I would say. I will remove it.